typescript/sdk/src/providers/SmartProvider/SmartProvider.ts

```diff
@@ -55,6 +55,36 @@ const DEFAULT_STAGGER_DELAY_MS = 1000; // 1 seconds
 
 type HyperlaneProvider = HyperlaneEtherscanProvider | HyperlaneJsonRpcProvider;
 
+/**
+ * Enum containing common error message phrases related to  transactions.
+ * These phrases are used to detect non-retryable errors when interacting with the RPCs.
+ */
+export enum TX_ERROR_MESSAGE_PHRASES {
+  Known = 'known',
+  Nonce = 'nonce',
+  Underpriced = 'underpriced',
+  InsufficientFunds = 'insufficient funds',
+  InsufficientBalance = 'insufficient balance',
+  Revert = 'revert',
+}
+export function hasNonRetryableError(params: { method: string; error: Error }) {
+  const { method, error } = params;
+  const TX_METHODS = ['sendRawTransaction', 'sendTransaction'];
```

Comment:
I would've thought these would need to include the `eth_` prefix, e.g. `eth_sendRawTransaction`, `eth_call`, etc